### ISSUE TITLE
Cannot logon

### ISSUE THREAD

**Issue by edmond25** [pettazz/pygooglevoice#21]
I type the correct password and user name but gvoice prompt me "Login failed. Retry?[Y/n]", I did using normally on 16-7. What happen??

**Comment by numericOverflow** [comment]
Same here - seems like something changed & broke.  My last successful SMS via pygooglevoice was July 20th, been getting login errors since.

**Comment by woodyl** [comment]
I've had the same problem and I've been working on it for a couple of days.  I think the problem might be caused by a change in the URLs that are used for login.  I notice that now the login process is in 2 stages; on the first page, you put in the email address and then you click 'Next' and then you put in the password on the 2nd page.  I'm not sure if it was like that before, but I'm thinking this is the change that's causing the problem.

I've been looking at the page source on the login pages, but I haven't been able to figure it out so far.  If anybody has any pointers for how to work this out, I'd be happy to experiment and look for a fix.

**Comment by twbarber** [comment]
This is most commonly caused by having Google two factor authentication enabled, which pygooglevoice does not support. Another cause could be Google swapping around request parameters on the login page  I'll take a look at this more tonight.

**Comment by numericOverflow** [comment]
@woodyl - I found a different version that works.  I don't have time now to dig in and unify their code with this branch, but looks like it's a mix of the "BASE" & LOGOUT vars in settings.py and also the "galx" & "service" vars in voice.py have changed yet again.

Look here for the working version I found for reference: [http://pbxinaflash.com/community/threads/sms-with-google-voice-is-back.19717/#post-122879](url)

--> use the code & apply the patch shown in that post

```
BASE = 'https://www.google.com/voice/'
LOGOUT = BASE + 'account/signout'
```
```
galx = re.search(r"type=\"hidden\" name=\"GALX\" value=\"(.+)\"", content).group(1)
service = re.search(r"type=\"hidden\" name=\"service\" value=\"(.+)\"", content).group(1) 
```

**Comment by twbarber** [comment]
@numericoverflow good catch!

**Comment by woodyl** [comment]
@twbarber  I don't think two factor authentication is the problem here.

**Comment by woodyl** [comment]
@numericOverflow - What are you using for the LOGIN = parameter?

**Comment by numericOverflow** [comment]
Here you go:
`
LOGIN = 'https://accounts.google.com/ServiceLoginAuth?service=grandcentral'
`

**Comment by woodyl** [comment]
Hmm..  Still not working for me.  The problem is that I've taken code from several sources and made a lot of changes.  Also, I rewrote some of the code to make it compatible with Python3.  I'll have to do some more digging to see why my setup is still not able to log in.

**Comment by woodyl** [comment]
Wait,  I looked at your LOGIN parameter again and saw that mine was:
LOGIN = 'https://accounts.google.com/ServiceLogi?service=grandcentral' 

I changed it to the one you're using and now it's working!  Thanks for your help!

**Comment by woodyl** [comment]
BTW, I get an error when I use this parameter:

```
LOGOUT = BASE + 'account/signout'

# this one works for me:
LOGOUT = 'https://www.google.com/voice/account/signout'
```

**Comment by edmond25** [comment]
I edit package settings.py
Change the "LOGIN" to following value:
LOGIN = 'https://accounts.google.com/ServiceLoginAuth?service=grandcentral'

Google Voice up again

**Comment by illgitthat** [comment]
Same as @edmond25 just changed the LOGIN variable in settings.py and it seems to be working fine now.

**Comment by chris001** [comment]
See #27 issue solved when you log into your google voice account then browse to the following page
https://accounts.google.com/DisplayUnlockCaptcha
>    **Allow access to your Google account**
>    As a security precaution, Google may require you to complete this additional step when signing into a new device or application.
>
>    To allow access, click the Continue button below.

**Comment by AdroitProgrammer** [comment]
Does this still work? I am trying to get this working in the C# version I don't understand how the fix applies to that code yet. Is there anyone who can help?

**Comment by chris001** [comment]
Yes it works. If you just take the current python master version, and translate it to c#, it should work fine.  Or, just go line by line and make sure all the corresponding variables in the c# version are the same values as the python version, and the logic the same.
